### ISSUE TITLE
How to cancel the automatic setting of the force sensor to 0 after the UR5e robot is started？

### ISSUE THREAD

**Issue by MZJR** [UniversalRobots/Universal_Robots_ROS_Driver#512]
Every time I re-run the external control program on the teach pendant, the wrench data is automatically zeroed(), no matter what pose the end of the arm is in. As we all know, the data after zeroing does not reflect the real force, and I want to calibrate myself through [force_torque_tools](https://github.com/kth-ros-pkg/force_torque_tools)
Would you like to help me to solve this? Thanks, and Best regards

# Versions
Ubuntu20.04 + Noetic

**Comment by fmauch** [comment]
@urrsk is the wrench data zeroed on each program start or would we have to take a look into the generated script code here?

**Comment by MZJR** [comment]
BTW,After zeroing, I let the end of the robotic arm rotate at a constant speed along the z-axis (the center of mass of the end fixture is on the z-axis), and obtained the force data of the xy-axis (blue is the x-axis, red is the y-axis). Is this a bug in the program acquisition or a bug in the force sensor itself?
![up置零有爪-6 18~6 18](https://user-images.githubusercontent.com/73015268/157818411-f01e1d51-1fed-4889-9e46-a575b597c3b0.png)

**Comment by urrsk** [comment]
> @urrsk is the wrench data zeroed on each program start or would we have to take a look into the generated script code here?

Yes, it is zeroed every time you start a program.
There exist a raw wrench channel on the RTDE protocol. I will suggest using the raw tool wrench if you want to make your own e.g. gravity compensation.
@MZJR Regarding the sinus error. This is an unwanted physical phenom in the sensor. If it is too high, you should contact your distributor/service for getting it corrected.

**Comment by MZJR** [comment]
> > @urrsk is the wrench data zeroed on each program start or would we have to take a look into the generated script code here?
> 
> Yes, it is zeroed every time you start a program. There exist a raw wrench channel on the RTDE protocol. I will suggest using the raw tool wrench if you want to make your own e.g. gravity compensation. @MZJR Regarding the sinus error. This is an unwanted physical phenom in the sensor. If it is too high, you should contact your distributor/service for getting it corrected.

Thanks for your reply.

**Comment by 159chenJie159** [comment]
Hello, based on my observation, the load compensation effect achieved by setting the payload and zero for UR5e and FT300 seems to be average — I'm not sure if this is reasonable.

With a 0.8 kg payload, changing the robot's posture results in an error of several Newtons.
So I tried reading the ft_raw_wrench for gravity compensation, but the values I got were very large, over 20,000. I'm not sure if this is correct or what the units are.
